azurekeyvault-flexvolume/oauth.go

```diff
@@ -84,7 +85,7 @@ type NMIResponse struct {
 }
 
 // GetKeyvaultToken retrieves a new service principal token to access keyvault
-func GetKeyvaultToken(grantType OAuthGrantType, cloudName, tenantID string, usePodIdentity, useVmManagedIdentity bool, vmManagedIdentityClientID, aADClientSecret, aADClientID, podname, podns string) (authorizer autorest.Authorizer, err error) {
+func GetKeyvaultToken(grantType OAuthGrantType, cloudName, tenantID string, usePodIdentity, useVmManagedIdentity bool, vmManagedIdentityClientID, aADClientSecret, aADClientID, podname, podns, nmiport string) (authorizer autorest.Authorizer, err error) {
```

Comment:
Since the parameter list is quite long, we should consider switching this to a struct. Not scoped for this PR, but adding a comment so we can update this later.